Modules/rat_command.py

```diff
@@ -85,10 +92,14 @@ class Commands:
             # someone didn't set me.
             raise CommandException(f"Bot client has not been created or not handed to Commands.")
 
+            # command didn't start with prefix, ignore it
+        if not message.startswith(cls.prefix):
+            cls.log.debug(f"caught a command without prefix: {message}")
+            raise PrefixInvalidException(f"I got an unknown prefix!'")
```

Comment:
Hey, good thing you added this check. (I completely forgot to reimplement that after I redid the command invocation)

However, raising an exception on expected behaviour is something that should generally be avoided. 

More importantly, this case will happen tons of times (on each message which isn't a command, likely more often than proper command invocations) and exceptions aren't the most performant thing in programming, especially when they are propagated all the way to the event loop (i.e. not caught by us anywhere) and printed, as would be the case here.

Long story short, this would likely grind Mecha to a snails pace, simply not doing anything / returning immediately would be preferable. (Doing this check in `MechaClient.on_message` would be the best solution in my opinion as it gets rid of an unnecessary function call, but that part is debatable)

Reply:
Yeah the initial code was to just return None, but unknown wanted an exception thrown. I preferred the first method although in my testing it only fires on non-alphanumeric characters as the first character like a command trigger (so *, ., + trigger it but normal text does not) so it won't fire as often. I'm fine with either method that makes sense.

I'll correct my git settings, thanks. This is the first ever project I've contributed to so I'm learning still.

Reply:
The way i read it is this condition should not be reached, as it shouldn't get past the on_message handler. Therefore this would not be a normal runtime condition and instead an error condition.
That said, the check for the command prefix should probably occur in on_message as that would be the first step in the command execution chain.